openeogeotrellis/deploy/batch_job.py

```diff
@@ -120,24 +119,56 @@ def extract_result_metadata(tracer: DryRunDataTracer) -> dict:
             logger.warning("Multiple aggregate_spatial geometries: {c}".format(c=len(aggregate_spatial_geometries)))
         agg_geometry = aggregate_spatial_geometries[0]
         if isinstance(agg_geometry, BaseGeometry):
+            # We only allow EPSG:4326 for the BaseGeometry case to keep things simple
+            # and prevent complicated problems with CRS transformations.
+            # The aggregation geometry comes from Shapely, but Shapely itself does not
+            # support coordinate system transformations.
+            # See also: https://shapely.readthedocs.io/en/stable/manual.html#coordinate-systems
+            bbox_crs = "EPSG:4326"
             bbox = agg_geometry.bounds
             geometry = mapping(agg_geometry)
-            area = area_in_square_meters(agg_geometry, "EPSG:4326")
+            area = area_in_square_meters(agg_geometry, bbox_crs)
         elif isinstance(agg_geometry, DelayedVector):
             bbox = agg_geometry.bounds
             # Intentionally don't return the complete vector file. https://github.com/Open-EO/openeo-api/issues/339
             geometry = mapping(Polygon.from_bounds(*bbox))
             area = agg_geometry.area
+            bbox_crs = agg_geometry.crs
         elif isinstance(agg_geometry, DriverVectorCube):
             bbox = agg_geometry.get_bounding_box()
             geometry = agg_geometry.get_bounding_box_geojson()
             area = agg_geometry.get_bounding_box_area()
+            bbox_crs = agg_geometry.get_crs()
         else:
             logger.warning(f"Result metadata: no bbox/area support for {type(agg_geometry)}")
 
+        # The aggregation geometries return tuples for their bounding box.
+        # Keep the end result consistent and convert it to a list.
+        if isinstance(bbox, tuple):
+            bbox = [*bbox]
```

Comment:
I think `bbox = list(bbox)`  is more idiomatic than `[*bbox]`  (I didn't even know this works)

Reply:
Done.